test/test_hydrostatic_free_surface_models.jl

```diff
@@ -97,26 +104,70 @@ end
 
     for arch in archs
         for topo in topos
-            @testset "Time-stepping HydrostaticFreeSurfaceModels [$arch, $topo]" begin
-                @info "  Testing time-stepping HydrostaticFreeSurfaceModels [$arch, $topo]..."
-                @test time_stepping_hydrostatic_free_surface_model_works(arch, topo, nothing)
+            grid = RegularRectilinearGrid(size=(1, 1, 1), extent=(1, 1, 1), topology=topo)
+
+            @testset "Time-stepping Rectilinear HydrostaticFreeSurfaceModels [$arch, $topo]" begin
+                @info "  Testing time-stepping Rectilinear HydrostaticFreeSurfaceModels [$arch, $topo]..."
+                @test time_step_hydrostatic_model_works(arch, grid)
+            end
+        end
+
+        rectilinear_grid = RegularRectilinearGrid(size=(1, 1, 1), extent=(1, 1, 1), halo=(3, 3, 3))
+        lat_lon_sector_grid = RegularLatitudeLongitudeGrid(size=(1, 1, 1), longitude=(0, 60), latitude=(15, 75), z=(-1, 0))
+        lat_lon_strip_grid = RegularLatitudeLongitudeGrid(size=(1, 1, 1), longitude=(-180, 180), latitude=(15, 75), z=(-1, 0))
```

Comment:
Free energy balance climate model 🌐

Reply:
ahahah it is a bit confusing to think about this "grid point"